tests/utils.py

```diff
@@ -351,3 +355,28 @@ def _application_docker_container(
         container.reload()
     yield container
     container.remove(force=True)
+
+
+def find_application_pid(pid: int) -> int:
+    # Application might be run using "sh -c ...", we detect the case and return the "real" application pid
+    process = Process(pid)
+    if (
+        process.cmdline()[0].endswith("sh")
+        and process.cmdline()[1] == "-c"
+        and len(process.children(recursive=False)) == 1
+    ):
+        pid = process.children(recursive=False)[0].pid
+
+    return pid
+
+
+def assert_jvm_flags_equal(actual_jvm_flags: Optional[List], expected_jvm_flags: List) -> None:
+    assert actual_jvm_flags is not None, "actual_jvm_flags is None"
+
+    assert len(actual_jvm_flags) == len(expected_jvm_flags), f"{actual_jvm_flags} != {expected_jvm_flags}"
+
+    for actual_flag_dict, expected_flag_dict in zip(actual_jvm_flags, expected_jvm_flags):
+        actual_flag_dict.pop("value")
+        expected_flag_dict.pop("value")
+
+        assert actual_flag_dict == expected_flag_dict, f"{actual_flag_dict} != {expected_flag_dict}"
```

Comment:
```suggestion
        assert actual_flag_dict == expected_flag_dict
```

didn't we agree that pytest does the pretty print for you? :sweat_smile: